File: Java/Adapter/src/com/frank/Adapter.java
package com.frank;

/**
 * Created by frank on 2015/12/27.
 */

/**
 *  适配器
 */

public class Adapter implements Target {

    public  Adapter(Adaptee adaptee) {
        this.adaptee = adaptee;
    }

    // 进行适配
    public void request(){
        this.adaptee.specificRequest();
    }

    // 持有被适配者对象
    private Adaptee adaptee;
}

File: C++/Adapter/Adapter.cpp
#include "Adapter.h"
#include <iostream>

Target::Target() {

}
Target::~Target() {

}

void Target::Request() { 
	std::cout<<"Target::Request"<<std::endl; 
}


Adaptee::Adaptee() {

}
Adaptee::~Adaptee() {

}
void Adaptee::SpecificRequest(){
	std::cout<<"Adaptee::SpecificRequest" <<std::endl;
}

Adapter::Adapter(Adaptee* ade) { 
	this->_ade = ade; 
}

Adapter::~Adapter() {

}

void Adapter::Request() { 
	_ade->SpecificRequest(); 
}